lib/controller.js

```diff
@@ -10,255 +10,260 @@ var updates = require('./update-fetch');
+            // Report the sadness
+            return reject(noTessels);
+          } else if (foundTessels.length === 1) {
             // Close all opened connections and resolve
             controller.closeTesselConnections(foundTessels)
               .then(resolve);
-          });
-      }
-    });
+          }
+          // If we have only one Tessel or two Tessels with the same name (just USB and LAN)
+          else if (foundTessels.length === 1 ||
+            (foundTessels.length === 2 && foundTessels[0].name === foundTessels[1].name)) {
+            // Close all opened connections and resolve
+            controller.closeTesselConnections(foundTessels)
+              .then(resolve);
+          }
+          // Otherwise
+          else {
+            logs.info('Multiple Tessels found.');
+            // Figure out which Tessel will be selected
+            return controller.runHeuristics(opts, foundTessels)
+              .then(function logSelected(tessel) {
+                // Report that selected Tessel to the user
+                logs.info('Will default to %s.', tessel.name);
+              })
+              .catch(function(err) {
+                if (!(err instanceof controller.HeuristicAmbiguityError)) {
+                  return controller.closeTesselConnections(foundTessels)
+                    .then(reject.bind(this, err));
+                }
+              })
+              .then(function() {
+                // Helpful instructions on how to switch
+                logs.info('Set default Tessel with environment variable (e.g. "export TESSEL=bulbasaur") or use the --name flag.');
+                // Close all opened connections and resolve
+                controller.closeTesselConnections(foundTessels)
+                  .then(resolve);
+              });
+          }
+        });
 
-    // Stop the search if CTRL+C is hit
-    process.once('SIGINT', function() {
-      // If the seeker exists (it should)
-      if (seeker !== undefined) {
-        // Stop looking for more Tessels
-        seeker.stop();
-      }
+        // Stop the search if CTRL+C is hit
+        process.once('SIGINT', function() {
+          // If the seeker exists (it should)
+          if (seeker !== undefined) {
+            // Stop looking for more Tessels
+            seeker.stop();
+          }
+        });
+      });
     });
-  });
 };
 
 Tessel.get = function(opts) {
-  return new Promise(function(resolve, reject) {
-    logs.info('Looking for your Tessel...');
-    // Collection variable as more Tessels are found
-    var tessels = [];
-
-    // Store the amount of time to look for Tessel in seconds
-    var seekerOpts = {
-      timeout: (opts.timeout || 2) * 1000,
-      usb: opts.usb,
-      lan: opts.lan,
-      authorized: true
-    };
-
-    if (opts.authorized !== undefined) {
-      seekerOpts.authorized = opts.authorized;
-    }
+  // check whether --key is set
+  return provision.setDefaultKey(opts.key)
```

Comment:
If this is a check, then `setDefaultKey` is an inappropriate name for the operation